core/src/test/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreSSLAuthTest.java

```
@@ -0,0 +1,94 @@
+import org.slf4j.LoggerFactory;
+
+import org.apache.kafka.common.errors.TimeoutException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class KafkaStoreSSLAuthTest extends SSLClusterTestHarness {
+  private static final Logger log = LoggerFactory.getLogger(KafkaStoreSSLAuthTest.class);
+
+  @Before
+  public void setup() {
+    log.debug("Zk conn url = " + zkConnect);
+  }
+
+  @After
+  public void teardown() {
+    log.debug("Shutting down");
+  }
+
+  @Test
+  public void testInitialization() {
+    KafkaStore<String, String> kafkaStore = StoreUtils.createAndInitSSLKafkaStoreInstance(zkConnect,
+            zkClient, clientSslConfigs, requireSSLClientAuth());
+    kafkaStore.close();
+  }
+
+  @Test(expected = TimeoutException.class)
+  public void testInitializationWithoutClientAuth() {
+    KafkaStore<String, String> kafkaStore = StoreUtils.createAndInitSSLKafkaStoreInstance(zkConnect,
+            zkClient, clientSslConfigs, false);
+    kafkaStore.close();
+
+    // TODO: make the timeout shorter so the test fails quicker.
+  }
+
+  @Test
+  public void testIncorrectInitialization() {
+    KafkaStore<String, String> kafkaStore = StoreUtils.createAndInitSSLKafkaStoreInstance(zkConnect,
+            zkClient, clientSslConfigs, requireSSLClientAuth());
+    try {
+      kafkaStore.init();
+      fail("Kafka store repeated initialization should fail");
+    } catch (StoreInitializationException e) {
+      // this is expected
+    }
+    kafkaStore.close();
+  }
+
+  @Test
+  public void testSimplePut() throws InterruptedException {
+    KafkaStore<String, String> kafkaStore = StoreUtils.createAndInitSSLKafkaStoreInstance(zkConnect,
+            zkClient, clientSslConfigs, requireSSLClientAuth());
+    String key = "Kafka";
+    String value = "Rocks";
+    try {
+      kafkaStore.put(key, value);
+    } catch (StoreException e) {
+      fail("Kafka store put(Kafka, Rocks) operation failed");
```

Instead of failing, maybe you can just propagate the exception? Otherwise, we won't be able to see the stacktrace in the output.